### DISCUSSION TITLE
Error when trying to save card created with Core Methods token

### DISCUSSION THREAD

**Discussion by Oteiza-a** [mercadopago/sdk-js#161] [General]
Hello, we are implementing the "Core Methods" integration at our checkout ([documentation](https://www.mercadopago.cl/developers/es/docs/checkout-api/integration-configuration/card/integrate-via-core-methods)) , and we have successfully been able to generate the card token through the `mp.fields.createCardToken` function. 

Here is an example of the request generated from our frontend by the `createCardToken` function:

<img width="462" alt="Screenshot 2023-06-08 at 16 02 56" src="https://github.com/mercadopago/sdk-js/assets/49501058/b01e54f0-306e-4739-acc9-7c42ea56e3f2">

The response:

<img width="461" alt="Screenshot 2023-06-08 at 16 04 36" src="https://github.com/mercadopago/sdk-js/assets/49501058/e47b1c7e-6fdd-44ad-bb3c-91ad6392cfde">

After that, we wanted to save this card, as shown in [this documentation](https://www.mercadopago.cl/developers/es/docs/checkout-api/cards-and-customers-management/create-customer-and-card), with the `mercadopago.card.create` function, but we were getting this error, even though the `token` and the `customer_id` were being sent correctly:

<img width="1162" alt="Screenshot 2023-06-08 at 15 51 38" src="https://github.com/mercadopago/sdk-js/assets/49501058/efbc0aa6-4a51-432f-9642-523beba47ea6">

So then we tried sending the `payment_method_id`, and the `issuer_id`, as suggested by the documentation, but now we are getting this error:

<img width="1458" alt="Screenshot 2023-06-08 at 15 55 55" src="https://github.com/mercadopago/sdk-js/assets/49501058/e80a626a-cbc7-4696-acee-8d95d9e4432f">

We have checked many times that everything is sent correctly, yet we keep getting that error. We have tried tokens generated with the documentation's [test cards](https://www.mercadopago.cl/developers/es/docs/checkout-pro/additional-content/test-cards) and with real cards and real names, yet the result is the same.

**Comment by lucmantovani** [comment]
Hello @Oteiza-a!

Which version of SDK NodeJS are you using?
Also, could you send the snippet you are using to save the card?

**Comment by Oteiza-a** [comment]
Hi @lucmantovani, thank you for responding.

In the frontend we are using the V2 of the sdk, we are loading the script via `https://sdk.mercadopago.com/js/v2`, and in the backend we are using the "mercadopago" library, with the "^1.5.12" version.

Here is a snippet of our backend code getting the customer, and sending it with the rest of the data to the create card function:

```
async const saveCard(
  cardInfo: SaveCardRequestDto,
  customerEmail: string,
): Promise<CommonCardResponse> {
  const { cardToken, paymentMethodId, issuerId } = cardInfo

  const mpCustomerSearch = await MercadoPago.customers.search({ qs: { email: customerEmail } })
  let mpCustomer = mpCustomerSearch?.results?.find((customer) => customer.email === customerEmail)

  if (!mpCustomer) {
    mpCustomer = await MercadoPago.customers.create({ email: customerEmail })
  }

  const createCardPayload: CreateCardRequestDTO = { 
    token: cardToken, 
    customer_id: mpCustomer.id,
    payment_method_id: paymentMethodId,
    issuer_id: issuerId
  }
  this.logger.info('request from mp', createCardPayload)
  const newCard = await MercadoPago.card.create(createCardPayload);

  // ...
}
```

It is worth noting too that we were already creating and saving cards before integrating the Core Methods in the frontend.

**Reply by lucmantovani** [reply]
Thanks for the information!
Just one more question about you being able to create and save cards before: were you using the same snippet of NodeJS in the backend?
Have you tried sending only `customer_id` and `token`?
```js
const createCardPayload: CreateCardRequestDTO = { 
  token: cardToken, 
  customer_id: mpCustomer.id,
}
```

Also, another try is to send the DTO as
```js
const createCardPayload: CreateCardRequestDTO = { 
  token: cardToken, 
  customer_id: mpCustomer.id,
  payment_method_id: paymentMethodId,
  issuer: {
    id: issuerId,
  }
}
```

**Reply by Oteiza-a** [reply]
Hi @lucmantovani, thank you for your response. 

Before implementing the Core Methods form in the frontend, we saved cards only with `customer_id` or `token`, and had no problem. We had the same code snippet but without `issuer_id` or `payment_method_id`.

After implementing the Core Methods, we were sending only `customer_id` or `token`, and got the first error "invalid card owner" as i showed in my first question. 

After that we tried adding the `issuer_id` and `payment_method_id` and got the "invalid parameters" error, with the "body must be a Json object" description.

Now we tried to implement the issuer structure that you suggested:
```
issuer: {
    id: issuerId
}
```

And got the old "invalid card owner" again.

What else should we try?

**Reply by lucmantovani** [reply]
What do you mean by "Before implementing the Core Methods"?
How were you creating the token before?

**Reply by Oteiza-a** [reply]
We used to generate the card token with the sdk function `mp.createCardToken`, sending the card data directly, as seen in [this documentation](https://github.com/mercadopago/sdk-js/blob/main/API/core-methods.md#mp-instancecreatecardtokencardtokenparams)

**Reply by lucmantovani** [reply]
Oh, I see!
Well, even though now you are using fields (which is correct!) the token creation should not be the problem here

Have you tried creating the token as usual and then saving the card directly calling the API?

**Reply by Oteiza-a** [reply]
@lucmantovani i finally found what the problem was, we were using an old MercadoPago API key in the frontend, i changed it for the new one and it worked. Sorry for making unrelated questions, but it was hard for me to identify the origin of the error.

Thanks a lot for your help!

**Reply by lucmantovani** [reply]
No problem!
Glad it worked!
